lib/services/access_request.go

```diff
@@ -275,7 +275,20 @@ func CalculateAccessCapabilities(ctx context.Context, clock clockwork.Clock, clt
 		caps.SuggestedReviewers = v.SuggestedReviewers
 	}
 
-	caps.RequireReason = v.requireReason
+	if v.requireReasonForAllRoles {
+		caps.RequireReason = true
+	} else if req.RequestableRoles {
+		caps.RequireReason, _, err = v.isReasonRequired(ctx, caps.RequestableRoles, nil)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	} else {
+		caps.RequireReason, _, err = v.isReasonRequired(ctx, caps.ApplicableRolesForResources, nil)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
```

Comment:
Can we extract this logic into a separate function, something like: 
```suggestion
func (v ...) calcRequireReasonCap(ctx context.Context, req types.AccessCapabilitiesRequest, caps types.AccessCapabilities) (bool, error) {
	if v.requireReasonForAllRoles {
		return true, nil
	}
	if req.RequestableRoles {
		return v.isReasonRequired(ctx, caps.RequestableRoles, nil)
	}
	return v.isReasonRequired(ctx, caps.ApplicableRolesForResources, nil)
}
```